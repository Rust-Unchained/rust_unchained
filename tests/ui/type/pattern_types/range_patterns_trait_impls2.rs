#![feature(pattern_types, rustc_attrs)]
#![feature(core_pattern_type)]
#![feature(core_pattern_types)]
#![allow(incomplete_features)]

//! check that pattern types can have local traits
//! implemented for them.

use std::pat::pattern_type;

type Y = pattern_type!(u32 is 1..);

impl Eq for Y {} //~ ERROR E0277

fn main() {}
