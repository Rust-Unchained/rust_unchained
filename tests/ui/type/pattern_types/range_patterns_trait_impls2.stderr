error[E0277]: can't compare `(u32) is 1..=` with `(u32) is 1..=`
  --> $DIR/range_patterns_trait_impls2.rs:13:13
   |
LL | impl Eq for Y {}
   |             ^ no implementation for `(u32) is 1..= == (u32) is 1..=`
   |
   = help: the trait `PartialEq` is not implemented for `(u32) is 1..=`
note: required by a bound in `Eq`
  --> $SRC_DIR/core/src/cmp.rs:LL:COL

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
